src/shared/experience/expSite.ts

```diff
@@ -179,11 +179,21 @@ export class ExperienceSite {
 
   public async getRemoteMetadata(): Promise<SiteMetadata | undefined> {
     if (this.metadataCache.remoteMetadata) return this.metadataCache.remoteMetadata;
+    // Check if there are special characters in the site name
+    const isInvalidSiteName = hasSpacesOrSpecialChars(this.siteName);
+    // const originalSiteName = this.siteName;
+    if (isInvalidSiteName) {
+      const updatedSiteName = replaceSpacesAndSpecialChars(this.siteName);
+      this.siteName = updatedSiteName;
+    }
     const result = await this.org
       .getConnection()
       .query<{ Name: string; LastModifiedDate: string }>(
         `SELECT Name, LastModifiedDate FROM StaticResource WHERE Name LIKE 'MRT_experience_%_${this.siteName}'`
       );
+    // Changing the site name back to original after fetching data from StaticResource so it may not break the flow elsewhere
+    // this.siteName = originalSiteName;
```

Comment:
@nrkruk If we uncomment this, it breaks the flow when we try to bring the site up in some cases where site name have special chars other than "-"

Reply:
Not sure what this means. Does this impact functionality in some way?

Reply:
@nrkruk  I was resetting the site name back to original after executing the query but when it tries to bring up the site it fails. So, basically we will get that site bundle but it won't bring up the site.

Also, should we do the site name updation inside the constructor itself? It's already replacing the space with underscore character.

```
  public constructor(org: Org, siteName: string) {
    this.org = org;
    this.siteDisplayName = siteName.trim();
    this.siteName = this.siteDisplayName.replace(' ', '_');
  }
```